tools/mirror_weights_on_hf.py

```diff
@@ -62,6 +68,77 @@ def update_hf_uploads_json(json_file_path, latest_kaggle_handles):
     print("Updated hf_uploaded_presets.json with newly uploaded handles.")
 
 
+def update_model_cards_on_hugging_face(presets):
+    kaggle_api = KaggleApi()
+    kaggle_api.authenticate()
+    for model, data in presets.items():
+        try:
+            kaggle_handle = data["kaggle_handle"].removeprefix("kaggle://")
+            owner = "keras"
+            model_slug = kaggle_handle.split("/")[1]
+            model_metadata = kaggle_api.get_model_with_http_info(
+                owner, model_slug
+            )
+            description = model_metadata[0]["description"]
+            usage = model_metadata[0]["instances"][0]["usage"].replace(
+                "${VARIATION_SLUG}", model
+            )
+            usage = re.sub(
+                r'\.from_preset\(".*?"\)', f'.from_preset("{model}")', usage
+            )
+            hf_usage = usage.replace(model, f"hf://keras/{model}")
+
+            print(f"Downloading model metadata from Kaggle: {model}")
+
+            # --- Construct Model Card Markup ---
+            model_card_markup = (
+                "---\nlibrary_name: keras-hub\n---\n"
+                + f"### Model Overview\n{description}\n\n"
+            )
+
+            # Add usage sections if `usage` is not empty
+            if usage:
+                model_card_markup += (
+                    f"### Example Usage\n{usage}\n\n"
+                    "## Example Usage with Hugging Face URI\n\n"
```

Comment:
Should this one have two `#`s which makes it a bigger header?

Reply:
Done!